study-builder/src/main/java/org/broadinstitute/ddp/studybuilder/task/BrainPrequalV2.java

```diff
@@ -0,0 +1,275 @@
+import org.broadinstitute.ddp.model.activity.definition.question.QuestionDef;
+import org.broadinstitute.ddp.model.activity.revision.RevisionMetadata;
+import org.broadinstitute.ddp.studybuilder.ActivityBuilder;
+import org.broadinstitute.ddp.util.ConfigUtil;
+import org.broadinstitute.ddp.util.GsonUtil;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.sqlobject.SqlObject;
+import org.jdbi.v3.sqlobject.customizer.Bind;
+import org.jdbi.v3.sqlobject.customizer.BindList;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+import org.jdbi.v3.sqlobject.statement.SqlUpdate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Task to migrate Brain prequal to v2
+ */
+public class BrainPrequalV2 implements CustomTask {
+
+    private static final Logger LOG = LoggerFactory.getLogger(BrainPrequalV2.class);
+    private static final String DATA_FILE = "patches/prequal-v2.conf";
+
+    private Path cfgPath;
+    private Config cfg;
+    private Config varsCfg;
+    private Config dataCfg;
+
+    @Override
+    public void init(Path cfgPath, Config studyCfg, Config varsCfg) {
+        this.cfgPath = cfgPath;
+        this.cfg = studyCfg;
+        this.varsCfg = varsCfg;
+
+        File file = cfgPath.getParent().resolve(DATA_FILE).toFile();
+        if (!file.exists()) {
+            throw new DDPException("Data file is missing: " + file);
+        }
+        this.dataCfg = ConfigFactory.parseFile(file).resolveWith(varsCfg);
+    }
+
+    @Override
+    public void run(Handle handle) {
+        LOG.info("Executing BrainPrequalV2 task...");
+
+        String activityCode = dataCfg.getString("activityCode");
+        String versionTag = dataCfg.getString("versionTag");
+        UserDto adminUser = handle.attach(JdbiUser.class).findByUserGuid(cfg.getString("adminUser.guid"));
+        StudyDto studyDto = handle.attach(JdbiUmbrellaStudy.class).findByStudyGuid(cfg.getString("study.guid"));
+        Instant timestamp = Instant.parse(dataCfg.getString("timestamp"));
+        String studyGuid = studyDto.getGuid();
+        long studyId = studyDto.getId();
+        long activityId = ActivityBuilder.findActivityId(handle, studyId, activityCode);
+
+        ActivityDto activityDto = handle.attach(JdbiActivity.class)
+                .findActivityByStudyGuidAndCode(studyGuid, activityCode)
+                .orElseThrow(() -> new DDPException("Could not find id for activity " + activityCode + " and study id " + studyGuid));
+
+        String reason = String.format(
+                "Update activity with studyGuid=%s activityCode=%s to versionTag=%s",
+                studyGuid, activityCode, versionTag);
+        RevisionMetadata meta = new RevisionMetadata(timestamp.toEpochMilli(), adminUser.getUserId(), reason);
+        SqlHelper helper = handle.attach(SqlHelper.class);
+        String selfOptionText = "I have been diagnosed with brain cancer.";
+        String promptText = "First, how will you be participating in the Brain Cancer Project? ";
```

Comment:
Sampath, Erin said that the Brain Cancer Project was renamed to Brain Tumor Project. Please, correct all the cases where "Brain Cancer Project" occurs in this activity, and use new name in the upcoming activities.

Reply:
Thanks Pavel. Does that mean any text like existing consent which has "brain cancer" should be updated ?

Reply:
No, only the project name.